src/extra/python/scripts/remove_certain_restart_and_data_files.py

```diff
@@ -62,13 +75,27 @@ def keep_only_certain_restart_files_data_dir(exp_object, max_num_files, interval
         for x in files_to_keep:
                files_to_remove.remove(x) 
 
+        first_to_be_removed = True
+        number_removed = 0
+        number_not_removed = 0
             #Then we remove them.
-        for entry in files_to_remove:
+        for entry in files_to_remove[1:-1]:
```

Comment:
This would benefit from an assert to make sure there is at least one element to loop over